### PR TITLE
add/getConfig

### PR DESCRIPTION
**Pull request by rgadon107**
@hellofromtonya PR#14 for the api function `getConfig()` is ready for code review.

### CODE REVIEW

**Comment by rgadon107** [comment]
@hellofromtonya Hey Tonya, I rewrote all the tests for PR#14. I constructed a model on paper, evaluating the 2 inputs (`$store_key`, and `$config`) and the expected output from `getConfig()`.  The revised tests don't track the originals. But they do cover each of the edge cases that I identified & their expected outcomes. I'd like my code for this PR reviewed again. Thanks.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	/**
+	 * Test should return empty array when store key does not exist.
+	 */
+	public function test_should_return_empty_array_when_key_does_not_exist() {
+		$config = (array) require CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
+		_the_store( '', $config );
+
+		$this->assertSame( [], getConfig( '' ) );
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: @rgadon107 For this test, we are checking if what happens when the key does not exist in the store. Here are some thoughts to improve this test:

1. There’s no need to add a config with no key.  It won’t store it in the store.
2. What happens if you text for the method name as a key?

Combining these two, you could rewrite this test to:

```php
public function test_should_return_empty_array_when_key_does_not_exist() {
     $this->assertSame( [], getConfig( ‘’ ) );
     $this->assertSame( [], getConfig( ‘this_key_does_not_exist’ ) );
     $this->assertSame( [], getConfig( __METHOD__ ) );
}
```

What do notice from this test?

1.  You don’t need to handle the store.
2.  You are testing multiple keys to ensure you get what you expect.  

In other words, you’re checking more than an empty string to ensure the behavior works with different combinations.  Why?  It’s possible that the code checks for an empty string and has a different behavior with a different string.  We check that the behavior is consistent in each of those edge cases.

Does that make sense?

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+		// Clean up _the_store.
+		_the_store( '', null, true );
+	}
+
+	/**
+	 * Test should return an empty array when store key and config are not stored.
+	 */
+	public function test_should_return_empty_array_when_key_and_config_not_stored() {
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: This test is redundant to the previous one.  The store can not retrieve a config unless the key is bound to the config.  Right?  That means the key has to exist in the store.  Therefore, you can delete this test.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php (hunk 1 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: @hellofromtonya This makes perfect sense. It avoids dealing directly with `_the_store()`. And it tests the expected behavior and conditions more broadly. I like the approach. I'll incorporate that in.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php (hunk 2 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: @hellofromtonya I realized when I drew up my models that this test would be redundant. You're right, there's no need to include the test. I'll remove it.

Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: @hellofromtonya Test removed as of commit `eefbf7c` (06-05-19).

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php (hunk 1 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: @hellofromtonya The test for `'test_should_return_empty_array_when_key_does_not_exist'` only works for the first assert. The second and third asserts throw the Exception in `_the_store`. I added the second and third asserts to the method `'test_should_throw_exception_when_config_does_not_exist'` and that test passed.

Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: @hellofromtonya All three possible cases are now tested: 

1) Given store key & config array keyed by the store key;
2) Given store key & no config. Exception thrown.
3) Given no store key & no config. Returns empty array. 

Can this PR be approved?

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+		getConfig( __METHOD__ );
+	}
+
+	/**
+	 * Test should return empty array when store key does not exist.
+	 */
+	public function test_should_return_empty_array_when_key_does_not_exist() {
+		$this->assertSame( [], getConfig( '' ) );
+
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: @rgadon107 Remove this empty line.  I know it's nit-picky.  But consistency is important.  It's good to build up your code quality muscles.  Formatting is part of code quality.

Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php]: @hellofromtonya Updated with commit `cddd429` on 06-07-19.
